packages/lib/constants.ts

```diff
@@ -32,7 +32,11 @@ export const SENDER_NAME = process.env.NEXT_PUBLIC_SENDGRID_SENDER_NAME || "Cal.
 // This is the URL from which all Cal Links and their assets are served.
 // Use website URL to make links shorter(cal.com and not app.cal.com)
 // As website isn't setup for preview environments, use the webapp url instead
-export const CAL_URL = new URL(WEBAPP_URL).hostname.endsWith(".vercel.app") ? WEBAPP_URL : WEBSITE_URL;
+// If it's a .vercel.app domain, keep it.
+// Else use the website url if defined and finally fallback to the webapp url
+export const CAL_URL = new URL(WEBAPP_URL).hostname.endsWith(".vercel.app")
+  ? WEBAPP_URL
+  : process.env.NEXT_PUBLIC_WEBSITE_URL || WEBAPP_URL;
```

Comment:
Avoid hard-coding `cal.com` as much as possible. In our hosted instance NEXT_PUBLIC_WEBSITE_URL is always set as `https://cal.com` so there should not be any problems here. Also this won't disturb self-hosts.